include/qdmi/client.h

```diff
@@ -119,20 +119,45 @@ enum QDMI_SESSION_PARAMETER_T {
    * parameter to be set.
    */
   QDMI_SESSION_PARAMETER_TOKEN = 0,
+  /**
+   * @brief `char*` (string) The URL to an authentication server that is used
+   * for the authentication via username and password.
+   * @details For authentication via username and password, the device
+   * documentation *must* document which of the three following parameters must
+   * be set:
+   * - @ref QDMI_SESSION_PARAMETER_AUTHURL
+   * - @ref QDMI_SESSION_PARAMETER_USERNAME
+   * - @ref QDMI_SESSION_PARAMETER_PASSWORD
+   *
+   * @par The URL is used to retrieve a token afterward used for every request
+   * to the device. The username and password are used to authenticate the user
+   * at the authentication server.
+   *
+   * @par The driver documentation *must* document if the implementation
+   * requires this parameter to be set.
```

Comment:
This is still overly specific and too strict in my opinion.
```suggestion
   * @brief `char*` (string) The URL to an authentication server used as part of
   * the authentication procedure.
   * @details This parameter might be used as part of an authentication scheme
   * where an API token is received from an authentication server. This may,
   * additionally, require a username and a password, which can be set via the
   * @ref QDMI_SESSION_PARAMETER_USERNAME and @ref QDMI_SESSION_PARAMETER_PASSWORD
   * parameters.
   *
   * @par The driver documentation *must* document when the implementation
   * requires this parameter to be set and which additional parameters need to
   * be set in case this authentication method is used.
```